das/daser.go

```diff
@@ -184,23 +186,25 @@ func (d *DASer) catchUpScheduler(ctx context.Context, checkpoint int64) {
 		case <-ctx.Done():
 			return
 		case job := <-d.jobsCh:
-			// spawn catchUp routine
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-
-				d.catchUp(ctx, job)
-				// TODO @renaynay: assumption that each subsequent job is to a higher height than the previous.
-				//  I don't see why that is *not* the case, though.
-				checkpoint = job.to
-			}()
+			// perform catchUp routine
+			result := d.catchUp(ctx, job)
+			// update checkpoint before checking error as the checkpoint
+			// value from result represents the last successfully DASed height
+			// of the routine's success
+			checkpoint = result.checkpoint
+			// exit routine if a catch-up job was unsuccessful
+			if result.err != nil {
+				log.Errorw("catch-up routine failed", "attempted range (from, to)", job.from,
+					job.to, "last successfully sampled height", result.checkpoint)
+				return
+			}
```

Comment:
Are we ok with the fact that in this case, the node has to be restarted to continue catching up? I am a bit nervous about this, but this can be improved further on.

Reply:
Yes. I am okay with this for now. We SHOULD, however, log that the user should try this because they won't implicitly understand that they should restart. Eventually we need to have some sort of error handling logic for these issues.

Reply:
I will make an issue and also update the log

Reply:
https://github.com/celestiaorg/celestia-node/issues/554